policy-controller/k8s/status/src/index.rs

```diff
@@ -0,0 +1,246 @@
+                return;
+            }
+        };
+        let accepting_servers: Vec<ResourceId> = self
+            .index
+            .read()
+            .servers
+            .iter()
+            .filter(|server| route_binding.selects_server(server))
+            .cloned()
+            .collect();
+        let parent_statuses = route_binding
+            .parents
+            .iter()
+            .map(|parent| {
+                let ParentReference::Server(parent_reference_id) = parent;
+
+                // Is this parent in the list of parents which accept
+                // the route?
+                let accepted = accepting_servers
+                    .iter()
+                    .any(|accepting_parent| accepting_parent == parent_reference_id);
+                let condition = if accepted {
+                    k8s::Condition {
+                        last_transition_time: k8s::Time(Utc::now()),
+                        message: "".to_string(),
+                        observed_generation: None,
+                        reason: "Accepted".to_string(),
+                        status: "True".to_string(),
+                        type_: "Accepted".to_string(),
+                    }
+                } else {
+                    k8s::Condition {
+                        last_transition_time: k8s::Time(Utc::now()),
+                        message: "".to_string(),
+                        observed_generation: None,
+                        reason: "NoMatchingParent".to_string(),
+                        status: "False".to_string(),
+                        type_: "Accepted".to_string(),
+                    }
+                };
+                gateway::RouteParentStatus {
+                    parent_ref: gateway::ParentReference {
+                        group: Some(POLICY_API_GROUP.to_string()),
+                        kind: Some("Server".to_string()),
+                        namespace: Some(parent_reference_id.namespace.clone()),
+                        name: parent_reference_id.name.clone(),
+                        section_name: None,
+                        port: None,
+                    },
+                    controller_name: STATUS_CONTROLLER_NAME.to_string(),
+                    conditions: vec![condition],
+                }
+            })
+            .collect();
+        let status = gateway::HttpRouteStatus {
+            inner: gateway::RouteStatus {
+                parents: parent_statuses,
+            },
+        };
+        let patch = serde_json::json!({
+            "apiVersion": POLICY_API_VERSION,
+            "kind": "HTTPRoute",
+            "name": route_id.name,
+            "status": status,
+        });
+
+        // Patch the HTTPRoute with its status.
+        let api = k8s::Api::<k8s::policy::HttpRoute>::namespaced(
+            self.client.clone(),
+            &route_id.namespace,
+        );
+        if let Err(error) = api
```

Comment:
does this need a timeout? or can we document the default k8s timeout?

Reply:
658f36223